ultralytics/engine/trainer.py

```diff
@@ -428,6 +428,7 @@ def _do_train(self):
                     self.loss = loss.sum()
                     if RANK != -1:
                         self.loss *= self.world_size
+                    self.loss_items = self.loss_items.detach() # detach to prevent memory leaks
```

Comment:
❗ **CRITICAL**: Calling `detach()` unconditionally on `self.loss_items` assumes it is always a `torch.Tensor`. In several trainers (e.g., detection) loss items are converted to Python floats or lists for logging, which would cause an `AttributeError` here (`list`/`float` has no `detach`). This change may break non-MPS or non-tensor use cases. Consider guarding the detach by type so it only runs when `self.loss_items` is actually a tensor.

**Suggested change:**
```suggestion
                    self.loss_items = self.loss_items.detach() if isinstance(self.loss_items, torch.Tensor) else self.loss_items  # detach to prevent memory leaks when tensor
```